src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/DiagnosticsTelemetryModule.cs

```diff
@@ -43,6 +46,48 @@ public DiagnosticsTelemetryModule()
         }
 
         /// <summary>
+        /// Gets or sets a value indicating whether or not the Health Heartbeat feature is disabled.
+        /// </summary>
+        public bool IsHeartbeatEnabled
+        {
+            get
+            {
+                return this.isHeartbeatEnabled;
+            }
+
+            set
+            {
+                this.isHeartbeatEnabled = value;
+                if (this.HeartbeatProvider != null)
+                {
+                    this.HeartbeatProvider.IsEnabled = this.isHeartbeatEnabled;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the delay between heartbeats.
+        /// </summary>
+        public TimeSpan HeartbeatInterval { get; set; }
```

Comment:
So this property is immutable and `IsHeartbeatEnabled` is mutable? Why not simply propagate it to the Heartbeat provider?

Reply:
Agreed, this should have been updated to behave similarly to 'IsHeartbeatEnabled'. And for the same reason I cite before, it is required in this module as well as in the HeartbeatProvider module as well.

Perhaps this is a great time to talk about promoting the HeartbeatProvider to the same place as DiagnosticsTelemetryModule? Would it make sense to keep the two separate, but initialized in the same 'hard coded' manner as the DiagnosticsTelemetryModule is today? This would avoid our need to have 2 of the same values shuffled from one module to the other. Alternatively, I could move the logic straight into the DiagnosticsTelemetryModule instead and do away with the HeartbeatProvider altogether, although that seems like too much coupling... thoughts?